common/cached/types.go

```diff
@@ -0,0 +1,103 @@
+package cached
+
+import (
+	"github.com/hyperledger/fabric/protos/common"
+	"github.com/hyperledger/fabric/protos/peer"
+	"sync"
+)
+
+type Block struct {
+	*common.Block
+	cachedEnvs     []*Envelope
+	cachedMetadata []*Metadata
+	metaMtx        sync.RWMutex
```

Comment:
if we use this lock, it means this struct used in parallel. to avoid waiting, as we just `unmarshal `? without modify data, so optimistic lock looks better?